### ISSUE TITLE
Unable to load application on production profile

### ISSUE THREAD

**Issue by victorv977** [jhipster/generator-jhipster#2932]
Hi All,

I just create a new application based on an existing yo-rc.json:

~~~json
{
  "generator-jhipster": {
    "baseName": "manager",
    "packageName": "com.xxx.manager",
    "packageFolder": "com/xx/manager",
    "authenticationType": "oauth2",
    "hibernateCache": "ehcache",
    "clusteredHttpSession": "no",
    "websocket": "spring-websocket",
    "databaseType": "sql",
    "devDatabaseType": "mysql",
    "prodDatabaseType": "mysql",
    "searchEngine": "no",
    "useSass": false,
    "buildTool": "gradle",
    "frontendBuilder": "gulp",
    "enableTranslation": true,
    "enableSocialSignIn": false,
    "rememberMeKey": "456fca32e9876be9b48b542343e321a3c5b800ac",
    "testFrameworks": [
      "cucumber",
      "protractor"
    ]
  }
}
~~~

Then, build the project:
`./gradlew -Pprod bootRepackage`

And run the app:
`java -jar vmaxmanager-0.0.1-SNAPSHOT.war`

The app runs fine:
<img width="1059" alt="screen-shot-2016-02-18-18-04-14" src="https://cloud.githubusercontent.com/assets/1454910/13161187/2251cfb2-d66a-11e5-806e-138a139e8b40.png">

 But when I run the app with profile prod:
`java -jar vmaxmanager-0.0.1-SNAPSHOT.war --spring.profiles.active=prod`

The app seem to run ok in the console:
<img width="1823" alt="screen-shot-2016-02-18-18-06-30" src="https://cloud.githubusercontent.com/assets/1454910/13161231/6a9dac32-d66a-11e5-8aa3-add503b82ece.png">

But, when I try to open the page, I got a 404:
<img width="1081" alt="screen-shot-2016-02-18-18-07-32" src="https://cloud.githubusercontent.com/assets/1454910/13161249/8f72a4ea-d66a-11e5-9b57-a6ea6a962a8f.png">

I have the same issue when I deploy to tomcat the `.war.original` file with `-Dspring.profiles.active=prod` flag

**Comment by jdubois** [comment]
I confirm the issue in our v2.x branch.
This is because the gulp.js build didn't generate the "dist" folder, where the minified client application is supposed to be generated.
@deepu105 @atomfrede as you are using the same Gradle + Gulp.js stack, don't you also have this issue?

**Comment by deepu105** [comment]
this was working well before, ill take a second look

Thanks & Regards,
Deepu

**Comment by atomfrede** [comment]
It works on master. Need to check it on 2.x branch.

**Comment by cbornet** [comment]
The gulp setup was fixed in #2686. It seems the branch doesn't include this fix.

**Comment by atomfrede** [comment]
I can have a look at it later and fix it on the maintenance branch

**Comment by deepu105** [comment]
may be we can do a backport and do a maintenance release

Thanks & Regards,
Deepu

On Fri, Feb 19, 2016 at 4:10 PM, Christophe Bornet <notifications@github.com

**Comment by jdubois** [comment]
Yes this looks like an important bug, a maintenance release would be necessary.

**Comment by victorv977** [comment]
Should we wait for a new release? This patch #2686 fix the issue? do you have a patch?

**Comment by atomfrede** [comment]
I will see to provide pr for the 2.x branch such that we can do a maintenance release asap.

**Comment by deepu105** [comment]
closed via #2943
